paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java

```diff
@@ -237,21 +255,66 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
                             + " You can create TEMPORARY table instead if you want to create the table of other connector.");
         }
 
+        Identifier identifier = toIdentifier(tablePath);
+        Map<String, String> logSystemOptions = registerLogSystem(identifier, options);
         // remove table path
         String specific = options.remove(PATH.key());
-        if (specific != null) {
+        if (specific != null || !logSystemOptions.isEmpty()) {
+            options.putAll(logSystemOptions);
             catalogTable = catalogTable.copy(options);
         }
 
+        boolean unRegisterLogSystem = false;
         try {
             catalog.createTable(
-                    toIdentifier(tablePath),
-                    FlinkCatalog.fromCatalogTable(catalogTable),
-                    ignoreIfExists);
+                    identifier, FlinkCatalog.fromCatalogTable(catalogTable), ignoreIfExists);
         } catch (Catalog.TableAlreadyExistException e) {
+            unRegisterLogSystem = true;
             throw new TableAlreadyExistException(getName(), tablePath);
         } catch (Catalog.DatabaseNotExistException e) {
+            unRegisterLogSystem = true;
             throw new DatabaseNotExistException(getName(), e.database());
+        } finally {
+            if (unRegisterLogSystem && !logSystemOptions.isEmpty()) {
+                unRegisterLogSystem(identifier, options);
+            }
+        }
+    }
+
+    private Map<String, String> registerLogSystem(
```

Comment:
Can this be in separate class? Maybe just in `LogStoreRegister`, a static method?